`@swc-node/register@1.8.0` `/esm-register` does not recognise `tsconfig.json` paths

I was playing with the latest and greatest and notice that when running via: 

```none
node --import @swc-node/register/esm-register src/index.ts
```

… `tsconfig.json` path aliases are not recognised, _e.g._

```json
{
  "compilerOptions": {
    "paths": {
      "@/*": ["./src/*"]
    }
  },
}
```

[minimal reproducible example](https://github.com/mylesj/issue-swc-node-1)

<details>
  <summary>Stack trace</summary>

```
node:internal/process/esm_loader:34
      internalBinding('errors').triggerUncaughtException(
                                ^
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/src/common' imported from /Users/myles/dev/issue-swc-node-1/src/index.ts
    at finalizeResolution (node:internal/modules/esm/resolve:264:11)
    at moduleResolve (node:internal/modules/esm/resolve:917:10)
    at defaultResolve (node:internal/modules/esm/resolve:1130:11)
    at nextResolve (node:internal/modules/esm/hooks:865:28)
    at resolve (file:///Users/myles/dev/issue-swc-node-1/node_modules/.pnpm/@swc-node+register@1.8.0_@swc+core@1.3.107_@swc+types@0.1.5_typescript@5.3.3/node_modules/@swc-node/register/esm/esm.mjs:44:12)
    at nextResolve (node:internal/modules/esm/hooks:865:28)
    at Hooks.resolve (node:internal/modules/esm/hooks:303:30)
    at MessagePort.handleMessage (node:internal/modules/esm/worker:196:24)
    at [nodejs.internal.kHybridDispatch] (node:internal/event_target:826:20)
    at exports.emitMessage (node:internal/per_context/messageport:23:28) {
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file:///src/common'
}
```

</details>

> _Also of note: I am aware that native ESM would typically expect file-extensions to be specified in the import, but including them doesn't help in this case - though I think the expectation of `swc` would be not to care either way?_

Well, TS paths resolution doesn't work and that's why I was trying `swc-node` to begin with:
```
❯ node --no-warnings=ExperimentalWarning --loader ts-node/esm ./apps/lambda-api/src/extract-schema.ts

node:internal/process/esm_loader:34
      internalBinding('errors').triggerUncaughtException(
                                ^
Error: Cannot find module '/Users/joaodias/the-mill/lambda-gaming.nosync/apps/lambda-api/src/schemas/schema' imported from /Users/joaodias/the-mill/lambda-gaming.nosync/apps/lambda-api/src/extract-schema.ts
```

--- a/.changeset/spotty-mails-relate.md
+++ b/.changeset/spotty-mails-relate.md
@@ -1,0 +1,5 @@
+---
+"@fake-scope/fake-pkg": patch
+---
+
+fix: convert fileUrl to path before compile, close #753

--- a/.changeset/spotty-mails-relate.md
+++ b/.changeset/spotty-mails-relate.md
@@ -1,5 +1,6 @@
 ---
-"@fake-scope/fake-pkg": patch
+'@fake-scope/fake-pkg': patch
 ---
 
 fix: convert fileUrl to path before compile, close #753
+fix: remove baseUrl from esm to keep module import specifier, cause it use tsc resolver.

--- a/package.json
+++ b/package.json
@@ -31,7 +31,7 @@
     "lint": "eslint -c ./.eslintrc.yml .",
     "test": "ava",
     "test:jest": "jest --config jest.config.js",
-    "test:module": "cross-env SWC_NODE_PROJECT=packages/integrate-module/tsconfig.json node --loader=@swc-node/register/esm packages/integrate-module/src/index.ts",
+    "test:module": "cross-env SWC_NODE_PROJECT=packages/integrate-module/tsconfig.json node --import=@swc-node/register/esm-register packages/integrate-module/src/index.ts",
     "version": "pnpm install && git add .",
     "postinstall": "husky install"
   },

--- a/packages/register/__test__/ts-compiler-options-to-swc-config.spec.ts
+++ b/packages/register/__test__/ts-compiler-options-to-swc-config.spec.ts
@@ -10,7 +10,7 @@ test('default values', (t) => {
   const filename = 'some-file.tsx'
   const swcConfig = tsCompilerOptionsToSwcConfig(options, filename)
   const expected = {
-    baseUrl: process.cwd(),
+    baseUrl: undefined,
     module: 'es6',
     sourcemap: false,
     experimentalDecorators: false,

--- a/packages/register/esm.mts
+++ b/packages/register/esm.mts
@@ -94,7 +94,7 @@ export const load: LoadFn = async (url, context, nextLoad) => {
   if (context.format === 'ts') {
     const { source } = await nextLoad(url, context)
     const code = typeof source === 'string' ? source : Buffer.from(source).toString()
-    const compiled = await compile(code, url, tsconfig, true)
+    const compiled = await compile(code, fileURLToPath(url), tsconfig, true)
     return {
       format: 'module',
       source: compiled,

--- a/packages/register/esm.mts
+++ b/packages/register/esm.mts
@@ -90,11 +90,17 @@ interface LoadResult {
 type LoadArgs = [url: string, context: LoadContext, nextLoad?: (...args: LoadArgs) => Promise<LoadResult>]
 type LoadFn = (...args: Required<LoadArgs>) => Promise<LoadResult>
 
+const tsconfigForSWCNode = {
+  ...tsconfig,
+  paths: undefined,
+  baseUrl: undefined,
+}
+
 export const load: LoadFn = async (url, context, nextLoad) => {
   if (context.format === 'ts') {
     const { source } = await nextLoad(url, context)
     const code = typeof source === 'string' ? source : Buffer.from(source).toString()
-    const compiled = await compile(code, fileURLToPath(url), tsconfig, true)
+    const compiled = await compile(code, fileURLToPath(url), tsconfigForSWCNode, true)
     return {
       format: 'module',
       source: compiled,

--- a/packages/register/read-default-tsconfig.ts
+++ b/packages/register/read-default-tsconfig.ts
@@ -130,7 +130,7 @@ export function tsCompilerOptionsToSwcConfig(options: ts.CompilerOptions, filena
             useBuiltins: true,
           }
         : undefined,
-    baseUrl: resolve(options.baseUrl ?? './'),
+    baseUrl: options.baseUrl ? resolve(options.baseUrl) : undefined,
     paths: Object.fromEntries(
       Object.entries(options.paths ?? {}).map(([aliasKey, aliasPaths]) => [
         aliasKey,

